nifi-nar-bundles/nifi-dbcp-services-bundle/nifi-postgres-dbcp-service/src/main/java/org/apache/nifi/service/PostgreSQLConnectionPool.java

```diff
@@ -0,0 +1,255 @@
+            .build();
+
+    public static final PropertyDescriptor POSTGRES_DATABASE_URL = new PropertyDescriptor.Builder()
+            .fromPropertyDescriptor(DATABASE_URL)
+            .dependsOn(CONNECTION_URL_FORMAT, FULL_URL)
+            .build();
+
+    public static final PropertyDescriptor DATABASE_NAME = new PropertyDescriptor.Builder()
+            .name("database-name")
+            .displayName("Database Name")
+            .description("The name of the database to connect.")
+            .required(false)
+            .dependsOn(CONNECTION_URL_FORMAT, PARAMETERS)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+
+    public static final PropertyDescriptor DATABASE_HOSTNAME = new PropertyDescriptor.Builder()
+            .name("database-hostname")
+            .displayName("Database Hostname")
+            .description("The hostname of the database to connect.")
+            .required(false)
+            .dependsOn(CONNECTION_URL_FORMAT, PARAMETERS)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+
+    public static final PropertyDescriptor DATABASE_PORT = new PropertyDescriptor.Builder()
+            .name("database-port")
+            .displayName("Database Port")
+            .description("The port of the database to connect.")
+            .required(false)
+            .dependsOn(CONNECTION_URL_FORMAT, PARAMETERS)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+
+    public static final PropertyDescriptor KERBEROS_AUTH = new PropertyDescriptor.Builder()
+            .name("kerberos-auth")
+            .displayName("Kerberos Authentication")
+            .description("Use Kerberos authentication.")
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .build();
+
+    public static final PropertyDescriptor POSTGRES_KERBEROS_USER_SERVICE = new PropertyDescriptor.Builder()
+            .fromPropertyDescriptor(KERBEROS_USER_SERVICE)
+            .dependsOn(KERBEROS_AUTH, "true")
+            .build();
+
+    public static final PropertyDescriptor POSTGRES_KERBEROS_CREDENTIALS_SERVICE = new PropertyDescriptor.Builder()
+            .fromPropertyDescriptor(KERBEROS_CREDENTIALS_SERVICE)
+            .dependsOn(KERBEROS_AUTH, "true")
+            .build();
+
+    public static final PropertyDescriptor POSTGRES_KERBEROS_PRINCIPAL = new PropertyDescriptor.Builder()
+            .fromPropertyDescriptor(KERBEROS_PRINCIPAL)
+            .dependsOn(KERBEROS_AUTH, "true")
+            .build();
+
+    public static final PropertyDescriptor POSTGRES_KERBEROS_PASSWORD = new PropertyDescriptor.Builder()
+            .fromPropertyDescriptor(KERBEROS_PASSWORD)
+            .dependsOn(KERBEROS_AUTH, "true")
+            .build();

```

Comment:
These properties should be removed together with `KerberosCredentialService` since `KerberosUserService` provides the same functionality.